.changeset/lovely-pianos-breathe.md

```diff
@@ -0,0 +1,40 @@
+---
+'astro': minor
+---
+
+Adds support for custom parsers to file loader
+
+For example, with a toml file of this format:
```

Comment:
```suggestion
 The `file()` loader now accepts a second argument that defines a `parser` function. This allows you to specify a language parser (e.g. `toml.parse` or `csv-parser`) to create a collection from a file's contents. The `file()` loader will automatically detect and parse JSON and YAML files with no need for a `parser`.
 
The following example defines a content collection `dogs` using an array of tables in a `.toml` file:
```
It's not sufficient to show an example and expect people to be able to focus on what the important information is. Options have to be explained so that people know how to use them.

I've just made some guesses based on reading the code and the examples, but you're the expert! We will need some kind of explanation like above to set the stage for the examples, but I'll need you to make sure it's accurate!

Also, I didn't see any tests for and couldn't find anything in the code that explains how CSV would be parsed. I don't know anything about this, but a quick search shows that there are a few libraries that are commonly used in Node for this. Does someone have to install one of these packages themselves and use it? Does it matter which one? I think this will be a use case people will be very interested in, so we should show an example of doing that too!